### PR TITLE
[FLINK-35755] Upgrade to flink-shaded 19.0

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

Upgrade flink-shaded version

## Brief change log

* Upgraded flink-shaded to 18.0 and then 19.0
* adapted the code to updated guava version
* replaces jsonpath dependency with a shaded version of it

## Verifying this change

Everything should work as before

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-dist/pom.xml [hunk 1]
```diff
@@ -52,6 +52,40 @@ under the License.
...
+				</exclusion>
+				<exclusion>
+					<groupId>com.fasterxml.jackson.datatype</groupId>
+					<artifactId>jackson-datatype-jdk8</artifactId>
+				</exclusion>
+				<exclusion>
+					<groupId>org.yaml</groupId>
+					<artifactId>snakeyaml</artifactId>
+				</exclusion>
+			</exclusions>
```
Comment (dawidwys) [inline flink-dist/pom.xml:88]: @snuyanzin Is that necessary with the upgraded `flatten` plugin in `flink-shaded`? I believe the reason for those exclusions was that `18.0` pulled in the unshaded dependencies incorrectly, isn't it?

File: flink-table/flink-table-planner/pom.xml [hunk 2]
```diff
@@ -353,19 +353,6 @@ under the License.
...
-							<pattern>com.fasterxml</pattern>
-							<!-- From flink-shaded-jackson -->
-							<shadedPattern>org.apache.flink.shaded.jackson2.com.fasterxml</shadedPattern>
-						</relocation>
-						<relocation>
-							<pattern>com.jayway</pattern>
-							<!-- From flink-table-runtime -->
-							<shadedPattern>org.apache.flink.table.shaded.com.jayway</shadedPattern>
-						</relocation>
-
```
Comment (dawidwys) [inline flink-table/flink-table-planner/pom.xml:368]: We use a shaded `jsonpath` version, which has those relocations applied already.

File: flink-dist/pom.xml (hunk 1 above)
Comment (snuyanzin) [reply flink-dist/pom.xml:88]: yes, you're right, need to remove it

Comment (dawidwys) [reply flink-dist/pom.xml:88]: will do

**Review by snuyanzin — APPROVED** [review]

**Pull request closed without merging**